Source/NSBrowser.m

```diff
@@ -3205,58 +3242,103 @@ - (void) _remapColumnSubviews: (BOOL)fromFirst
   if (fromFirst)
     {
       for (i = _firstVisibleColumn; i <= _lastVisibleColumn; i++)
-        {
-          bc = [_browserColumns objectAtIndex: i];
-          sc = [bc columnScrollView];
-          [self addSubview: sc];
-
-          if ([bc isLoaded] && [bc columnMatrix] == firstResponder)
-            {
-              [_window makeFirstResponder: firstResponder];
-              setFirstResponder = YES;
-            }
-        }
+	{
+	  bc = [_browserColumns objectAtIndex: i];
+	  sc = [bc columnScrollView];
+	  [self addSubview: sc];
+
+	  if ([bc isLoaded] && [bc columnMatrix] == firstResponder)
+	    {
+	      [_window makeFirstResponder: firstResponder];
+	      setFirstResponder = YES;
+	    }
+	}
 
       if (firstResponder && setFirstResponder == NO)
-        {
-          [_window makeFirstResponder:
-                     [[_browserColumns objectAtIndex: _firstVisibleColumn]
-                       columnMatrix]];
-        }
+	{
+	  [_window makeFirstResponder:
+		     [[_browserColumns objectAtIndex: _firstVisibleColumn]
+		       columnMatrix]];
+	}
     }
   else
     {
       for (i = _lastVisibleColumn; i >= _firstVisibleColumn; i--)
-        {
-          bc = [_browserColumns objectAtIndex: i];
-          sc = [bc columnScrollView];
-          [self addSubview: sc];
-
-          if ([bc isLoaded] && [bc columnMatrix] == firstResponder)
-            {
-              [_window makeFirstResponder: firstResponder];
-              setFirstResponder = YES;
-            }
-        }
+	{
+	  bc = [_browserColumns objectAtIndex: i];
+	  sc = [bc columnScrollView];
+	  [self addSubview: sc];
+
+	  if ([bc isLoaded] && [bc columnMatrix] == firstResponder)
+	    {
+	      [_window makeFirstResponder: firstResponder];
+	      setFirstResponder = YES;
+	    }
+	}
 
       if (firstResponder && setFirstResponder == NO)
-        {
-          [_window makeFirstResponder:
-                     [[_browserColumns objectAtIndex: _lastVisibleColumn]
-                       columnMatrix]];
-        }
+	{
+	  [_window makeFirstResponder:
+		     [[_browserColumns objectAtIndex: _lastVisibleColumn]
+		       columnMatrix]];
+	}
     }
 }
 
+- (id) _itemForColumn: (NSInteger)column
+{
+  id item = nil;
+
+  if (column == 0)
+    {
+      item = [_browserDelegate rootItemForBrowser: self];
+    }
+  else
+    {
+      NSInteger col = column > 0 ? column - 1 : column;
+      NSBrowserColumn *bc;
+
+      bc = [_browserColumns objectAtIndex: col];
+      if (bc != nil)
+	{
+	  NSMatrix *matrix;
+
+	  matrix = [bc columnMatrix];
+	  if (matrix != nil)
+	    {
+	      NSArray *selectedCells = [matrix selectedCells];
+	      if (selectedCells != nil && [selectedCells count] > 0)
```

Comment:
The whole logic of this method seems strange to me. What is your intention here?
This will work for some very common cases, for example when a new column needs to be loaded because we double click on an item in the column before. But I am not sure that every time we need to load a new column there is always exactly one item selected in the previous column.

Reply:
The idea is to find the item that needs to be loaded for a given column.   When multiple items are selected it will  show nothing on macOS.  I will make changes to make this code behave similarly.  See below this screenshot from macOS:

![multiple_selection_macos](https://github.com/user-attachments/assets/4c061181-49c0-45c0-9509-df86558f3ce5)

Reply:
The implementation does not match the description you give above. For multiple selections the first gets used.